components/file_system/src/thread_io.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::borrow::BorrowMut;
+use std::cell::RefCell;
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+use std::sync::MutexGuard;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use nix::unistd::getpid;
+use nix::unistd::gettid;
+use nix::unistd::Pid;
+use strum::EnumCount;
+use thread_local::ThreadLocal;
+
+use crate::IOBytes;
+use crate::IOType;
+
+lazy_static! {
+    static ref THREAD_IO_SENTINEL_VEC: ThreadLocal<Arc<ThreadIOSentinel>> = ThreadLocal::new();
```

Comment:
why this be thread local?

Reply:
The io bytes data is recorded in `ThreadIOSentinel` so we need that in each thread. `thread_local::ThreadLocal` provide an iterator to access data in each thread so with that one thread could fetch data for every thread.